src/cdk-experimental/listbox/listbox.ts

```diff
@@ -0,0 +1,138 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.dev/license
+ */
+
+import {
+  computed,
+  contentChildren,
+  Directive,
+  ElementRef,
+  inject,
+  input,
+  model,
+  OnDestroy,
+  signal,
+} from '@angular/core';
+import {OptionPattern} from '../ui-patterns/listbox/option';
+import {ListboxInputs, ListboxPattern} from '../ui-patterns/listbox/listbox';
+import {Directionality} from '@angular/cdk/bidi';
+import {startWith, takeUntil} from 'rxjs/operators';
+import {Subject} from 'rxjs';
+
+@Directive({
+  selector: '[cdkListbox]',
+  exportAs: 'cdkListbox',
+  host: {
+    'role': 'listbox',
+    'class': 'cdk-listbox',
+    '[attr.tabindex]': 'state.tabindex()',
+    // '[attr.aria-disabled]': 'state.disabled()',
+    '[attr.aria-multiselectable]': 'state.multiselectable()',
+    '[attr.aria-activedescendant]': 'state.activedescendant()',
+    '[attr.aria-orientation]': 'state.orientation()',
+    '(focusin)': 'state.onFocus()',
+    '(keydown)': 'state.onKeydown($event)',
+    '(mousedown)': 'state.onMousedown($event)',
+    // '(focusout)': '_handleFocusOut($event)',
+    // '(focusin)': '_handleFocusIn()',
+  },
+})
+export class CdkListbox implements ListboxInputs, OnDestroy {
+  /** Whether the list is vertically or horizontally oriented. */
+  orientation = input<'vertical' | 'horizontal'>('vertical');
+
+  /** Whether multiple items in the list can be selected at once. */
+  multiselectable = input<boolean>(false);
+
+  /** Whether focus should wrap when navigating. */
+  wrap = input<boolean>(true);
+
+  /** Whether disabled items in the list should be skipped when navigating. */
+  skipDisabled = input<boolean>(true);
+
+  /** The focus strategy used by the list. */
+  focusStrategy = input<'roving tabindex' | 'activedescendant'>('roving tabindex');
+
+  /** The selection strategy used by the list. */
+  selectionStrategy = input<'follow' | 'explicit'>('follow');
+
+  /** The amount of time before the typeahead search is reset. */
+  delay = input<number>(0.5);
+
+  /** The ids of the current selected items. */
+  selectedIds = model<string[]>([]);
```

Comment:
IMO the listbox should have a `value` input as its model, nd tracking of the selected option IDs should be an implementation detail

Reply:
I'll leave this for a future PR. There will need to be a follow-up PR for the cdk-experimental/listbox because the public api still needs to be finalized